test/src/SpendPermissions/spendWithWithdraw.t.sol

```diff
@@ -456,4 +462,123 @@ contract SpendWithWithdrawTest is SpendPermissionManagerBase {
+        vm.assume(spender != address(0));
+        vm.assume(spender != address(account));
+        vm.assume(encodedSpender != spender); // Ensure encoded spender is different
+        vm.assume(start > 0);
+        vm.assume(end > 0);
+        vm.assume(start < end);
+        vm.assume(period > 0);
+        vm.assume(spend > 0);
+        vm.assume(allowance > 0);
+        vm.assume(allowance >= spend);
+
+        SpendPermissionManager.SpendPermission memory spendPermission = SpendPermissionManager.SpendPermission({
+            account: address(account),
+            spender: spender,
+            token: NATIVE_TOKEN,
+            start: start,
+            end: end,
+            period: period,
+            allowance: allowance,
+            salt: salt,
+            extraData: extraData
+        });
+
+        // Create withdraw request with incorrect encoded spender
+        MagicSpend.WithdrawRequest memory withdrawRequest = _createWithdrawRequest(encodedSpender);
+        withdrawRequest.amount = spend;
+        withdrawRequest.signature = _signWithdrawRequest(address(account), withdrawRequest);
+
+        vm.deal(address(magicSpend), allowance);
+
+        bytes memory signature = _signSpendPermission(spendPermission, ownerPk, 0);
+
+        vm.warp(start);
+
+        vm.startPrank(spender);
+        mockSpendPermissionManager.approveWithSignature(spendPermission, signature);
+
+        vm.expectRevert(
+            abi.encodeWithSelector(
+                SpendPermissionManager.InvalidWithdrawRequestSpender.selector, encodedSpender, spender
+            )
+        );
+        mockSpendPermissionManager.spendWithWithdraw(spendPermission, spend, withdrawRequest);
+        vm.stopPrank();
+    }
+
+    function test_spendWithWithdraw_success_withEncodedSpender(
+        address spender,
+        uint48 start,
+        uint48 end,
+        uint48 period,
+        uint160 allowance,
+        uint256 salt,
+        bytes memory extraData,
+        uint160 spend
+    ) public {
+        vm.assume(spender != address(0));
+        vm.assume(spender != address(account));
+        vm.assume(spender != address(magicSpend));
+        assumePayable(spender);
+        vm.assume(start > 0);
+        vm.assume(end > 0);
+        vm.assume(start < end);
+        vm.assume(period > 0);
+        vm.assume(spend > 0);
+        vm.assume(allowance > 0);
+        vm.assume(allowance >= spend);
+
+        SpendPermissionManager.SpendPermission memory spendPermission = SpendPermissionManager.SpendPermission({
+            account: address(account),
+            spender: spender,
+            token: NATIVE_TOKEN,
+            start: start,
+            end: end,
+            period: period,
+            allowance: allowance,
+            salt: salt,
+            extraData: extraData
+        });
+
+        // Create withdraw request with correctly encoded spender
+        MagicSpend.WithdrawRequest memory withdrawRequest = _createWithdrawRequest(spender);
```

Comment:
Fuzzing would be a bit more complete if we also had a `uint96` argument for the simulated randomness we'll be doing offchain. Would add something like
```suggestion
        // Create withdraw request with correctly encoded spender
        MagicSpend.WithdrawRequest memory withdrawRequest = _createWithdrawRequest(spender);
        withdrawRequest.nonce = (nonceEntropy << 160) | withdrawRequest.nonce;
```